src/protagonist/DLCS.Repository/Assets/AssetQueryX.cs

```diff
@@ -63,4 +64,48 @@ private static LambdaExpression CreateExpression(Type type, string propertyName)
 
         return Expression.Lambda(body, param);
     }
+
+    public static IQueryable<Asset> ApplyAssetFilter(this IQueryable<Asset> queryable, 
+        AssetFilter? assetFilter, bool filterOnSpace = false)
+    {
+        if (assetFilter == null)
+        {
+            return queryable;
+        }
+
+        var filtered = queryable;
+        // is HasText right here? Would we want to select string1 = ""?
```

Comment:
`HasText` makes sense, although I can see the value of explicitly wanting empty string. Could update later if it's needed.